app/controllers/application_controller/performance.rb

```diff
@@ -696,8 +696,13 @@ def perf_gen_data_before_wait
                              from_dt,
                              to_dt,
                              interval_type]
-      elsif %w(MiddlewareServer MiddlewareDatasource).include?(@perf_record.class.name.demodulize)
-        rpt = perf_get_chart_rpt("vim_perf_#{interval_type}_#{@perf_record.class.name.demodulize.underscore}")
+      elsif %w(MiddlewareServer MiddlewareDatasource MiddlewareMessaging).include?(@perf_record.class.name.demodulize)
+        report_name = if @perf_record.respond_to?(:chart_report_name)
+                        @perf_record.chart_report_name
```

Comment:
In the chart_report_name I guess that is where now it set the messaging reports.
I would try to think if we can unify that for MiddlewareServer and MiddlewareDatasource.
The motivation was that for a single entity (MiddlewareMessaging) we need to create several reports, am I right ?